programs/mango-v4/src/state/mango_account_components.rs

```diff
@@ -587,129 +786,145 @@ mod tests {
         assert_eq!(pos.avg_entry_price(&market), 10.0);
         assert_eq!(pos.break_even_price(&market), 10.0);
-        assert_eq!(pos.realized_pnl_native, I80F48::from(0));
+        assert_eq!(pos.realized_trade_pnl_native, I80F48::from(0));
+        assert_eq!(pos.settle_pnl_limit_realized_trade, 0);
     }
 
     #[test]
     fn test_quote_entry_short_increasing_from_zero() {
-        let mut market = PerpMarket::default_for_tests();
+        let mut market = test_perp_market(10.0);
         let mut pos = create_perp_position(&market, 0, 0);
         // Go short 10 @ 10
         pos.record_trade(&mut market, -10, I80F48::from(100));
         assert_eq!(pos.quote_running_native, 100);
         assert_eq!(pos.avg_entry_price(&market), 10.0);
         assert_eq!(pos.break_even_price(&market), 10.0);
-        assert_eq!(pos.realized_pnl_native, I80F48::from(0));
+        assert_eq!(pos.realized_trade_pnl_native, I80F48::from(0));
+        assert_eq!(pos.settle_pnl_limit_realized_trade, 0);
     }
 
     #[test]
     fn test_quote_entry_long_increasing_from_long() {
-        let mut market = PerpMarket::default_for_tests();
+        let mut market = test_perp_market(10.0);
         let mut pos = create_perp_position(&market, 10, -100);
         // Go long 10 @ 30
         pos.record_trade(&mut market, 10, I80F48::from(-300));
         assert_eq!(pos.quote_running_native, -400);
         assert_eq!(pos.avg_entry_price(&market), 20.0);
         assert_eq!(pos.break_even_price(&market), 20.0);
-        assert_eq!(pos.realized_pnl_native, I80F48::from(0));
+        assert_eq!(pos.realized_trade_pnl_native, I80F48::from(0));
+        assert_eq!(pos.settle_pnl_limit_realized_trade, 0);
     }
 
     #[test]
     fn test_quote_entry_short_increasing_from_short() {
-        let mut market = PerpMarket::default_for_tests();
+        let mut market = test_perp_market(10.0);
         let mut pos = create_perp_position(&market, -10, 100);
         // Go short 10 @ 10
         pos.record_trade(&mut market, -10, I80F48::from(300));
         assert_eq!(pos.quote_running_native, 400);
         assert_eq!(pos.avg_entry_price(&market), 20.0);
         assert_eq!(pos.break_even_price(&market), 20.0);
-        assert_eq!(pos.realized_pnl_native, I80F48::from(0));
+        assert_eq!(pos.realized_trade_pnl_native, I80F48::from(0));
+        assert_eq!(pos.settle_pnl_limit_realized_trade, 0);
     }
 
     #[test]
     fn test_quote_entry_long_decreasing_from_short() {
-        let mut market = PerpMarket::default_for_tests();
+        let mut market = test_perp_market(10.0);
         let mut pos = create_perp_position(&market, -10, 100);
         // Go long 5 @ 50
         pos.record_trade(&mut market, 5, I80F48::from(-250));
         assert_eq!(pos.quote_running_native, -150);
         assert_eq!(pos.avg_entry_price(&market), 10.0); // Entry price remains the same when decreasing
         assert_eq!(pos.break_even_price(&market), -30.0); // Already broke even
```

Comment:
Already broke even - I dont get this comment, isnt the user making even further more loss?